adal/authentication_context.py

```diff
@@ -67,8 +70,20 @@ def __init__(self, authority, validate_authority=None, cache=None):
         self._oauth2client = None
         self.correlation_id = None
         env_value = os.environ.get('ADAL_PYTHON_SSL_NO_VERIFY')
+        if api_version is not None:
+            warnings.warn(
+                """The default behavior of including api-version=1.0 on the wire
+                is now deprecated.
+                Future version of ADAL will change the default value to None.

```

Comment:
Yeah that is the tricky part which is not yet set on stone.

When we choose to maintain the backward compatibility of previous `api-version=1.0` behavior by default, from now on the majority of current customers and new customers (those whom got bitten by issue #54 and/or whom be persuaded by the warning message in this PR), would then have to do `AuthenticationContext(..., api_version=None)` explicitly. And this situation will last for quite a while until the next major version bump on ADAL. So lots of customers will be on the `api_version=None` side. By the time, we will have 2 options moving forward:

1. We may choose to completely remove the `api_version=...` parameter with a major version bump. That would be a HARD breaking change, because all customers who follow our advice this time would then have to change their code again to remove the `api_version=None`.

2. Or we may choose to keep the `api_version=...` parameter in the API signature, we just will not use it anymore, i.e. it will become a NO-OP. That way it will still be a subtle breaking change (and also shipped with a major version bump), but current `AuthenticationContext(..., api_version=None)` customers would not need to change their code.

All these sound more complicated than [simply remove api_version completely and immediately](https://github.com/AzureAD/azure-activedirectory-library-for-python/pull/56/files), but this is the price to pay for the backward compatibility.